builtin/providers/aws/resource_aws_elb.go

```diff
@@ -725,44 +725,61 @@ func resourceAwsElbUpdate(d *schema.ResourceData, meta interface{}) error {
 
 		removed := expandStringList(os.Difference(ns).List())
 		added := expandStringList(ns.Difference(os).List())
+		subnetconn := meta.(*AWSClient).ec2conn
 
-		if len(removed) > 0 {
-			detachOpts := &elb.DetachLoadBalancerFromSubnetsInput{
-				LoadBalancerName: aws.String(d.Id()),
-				Subnets:          removed,
+		// This section sorts the removed/added subnets based on availability zones
+		dup_az_count := 0
+		for i, r_subnet_id := range removed {
+			r_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
+				SubnetIds: []*string{r_subnet_id},
+			})
+			if err != nil {
+				return err
+			}
+			for j, a_subnet_id := range added {
+				a_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
```

Comment:
So there are some nuances here that took awhile to figure out. First, let's say our elb has two subnets in az1 and az2, and we want to replace them with different subnets in az1 and az2. If we first remove both subnets we get an error saying we can't remove all subnets from our elb. If we try and add the new subnets and then remove the old ones, we get an error saying we can't add subnets in the same AZ. So we have to go one by one removing and then adding. We could make have less calls by removing and adding n-1 subnets and then doing a final call for the last subnets but that sounds like overkill